### PR TITLE
Drop PHP 5.4 support

### PR DESCRIPTION
**Pull request by foxycode**
I'd like to upgrade upgrade `symfony/console` at least to version 2.4 to be able to use short array syntax. Unfortunately 2.4 minimum php version is `>=5.3.3`, so php 5.4 support needs to be dropped. 

Maybe I could look on `symfony/console` changes and try upgrade to version 3.0 insted. Are you interested?

### CODE REVIEW

**Comment by Majkl578** [comment]
I don't understand. You want to drop 5.4 support (why btw., what do you need from 5.5/5.6 you can't use in 5.4?), but then you add `>=5.3.3` to composer.json.

**Comment by foxycode** [comment]
@Majkl578 As I said, I'd like to use short array syntax, which is supported from php 5.4.. Currently supported `symfony/console` 2.3 wants php `>=5.3.3`. Next version 2.4 want php `>=5.3.3`. Hope I explained it right.

Maybe it will be ok to keep `symfony/console` at version 2.4 and only rise minimum version of `kdyby/console` to `>=5.4.0`, but I really don't have php4 to test it.

Anyway, php4 is no longer supported and php5 will be only till 1. July 2016, so I really think this is no problem for master branch when `symfony/console` current version is 3.0 and `nette/di` from master wants php `>=5.6.0`.

This is just a start, I want to make more PR's and if @fprochazka is ok with it, finally upgrade to `synfony/console` 3.0.

**Comment by foxycode** [comment]
@Majkl578 Now I see it. I was confused by php version strings :smile: What I really want to is upgrade to `symfony/console 3.0` which wants `php >=5.5.9`.

@fprochazka I'll rework my PR, will you merge it then?

**Comment by Majkl578** [comment]
I'm :+1: on adding support for 3.0, but :-1: on dropping 2.* support.
I don't have strong opinion about PHP version, I feel that 5.4 support doesn't hurt anyone though.

**Comment by fprochazka** [comment]
I haven't read the entire discussion yet, but I'm against dropping 5.4 unless absolutely neccesary.

**Comment by foxycode** [comment]
@fprochazka You already have `2.3` a `2.4` versions in `Kdyby/Doctrine`. So what about creating new `3.0` release of `Kdyby/Console` and keep the `2.x` version? What do you want me to do to achieve that? PR to another branch?

File: composer.json [hunk 1]
```diff
@@ -17,9 +17,10 @@
 		"issues": "https://github.com/kdyby/console/issues"
 	},
 	"require": {
+		"php": ">=5.3.3",
```
Comment (fprochazka) [inline composer.json]: this can be `>=5.4`

**Comment by fprochazka** [comment]
I've merged the short array conversion and php min version https://github.com/Kdyby/Console/commit/bc58d9010f831a2c8041d85c3abe1360314b587a.

**Comment by fprochazka** [comment]
Supporting symfony/console `3.0` will require a little more work. Also fixing kdyby/events will be probably neccesary.

**Comment by foxycode** [comment]
@fprochazka Fixing in what way?

**Comment by fprochazka** [comment]
I've tried just to remember why I didn't do it sooner

```patch
diff --git i/composer.json w/composer.json                                                                                                                                                                                                            
index 30d5680..c761014 100644                                                                                                                                                                                                                         
--- i/composer.json                                                                                                                                                                                                                                   
+++ w/composer.json                                                                                                                                                                                                                                   
@@ -20,7 +20,7 @@                                                                                                                                                                                                                                     
                "php": ">=5.4",                                                                                                                                                                                                                       
                "nette/di": "~2.3@dev",                                                                                                                                                                                                               
                                                                                                                                                                                                                                                      
-               "symfony/console": "~2.3"                                                                                                                                                                                                             
+               "symfony/console": "~2.3|~3.0"                                                                                                                                                                                                        
        },                                                                                                                                                                                                                                            
        "require-dev": {                                                                                                                                                                                                                              
                "nette/application": "~2.3@dev",                                                                                                                                                                                                      
@@ -47,7 +47,7 @@                                                                                                                                                                                                                                     
                                                                                                                                                                                                                                                      
                "nette/tester": "~1.6@rc",                                                                                                                                                                                                            
                "kdyby/events": "~2.4@dev",                                                                                                                                                                                                           
-               "symfony/event-dispatcher": "~2.3"                                                                                                                                                                                                    
+               "symfony/event-dispatcher": "~2.3|~3.0"
        },
        "suggest": {
                "nette/application": "For hooking the extension into the app and enabling running it through www/index.php",
```

and after some minor adjustments, this popped out

![snimek obrazovky porizeny 2016-03-15 21 47 11](https://cloud.githubusercontent.com/assets/158625/13793520/8019b7bc-eaf7-11e5-8c63-04a8ddd6b605.png)

**Pull request closed without merging**